pkg/auth/requests/authenticate.go

```diff
@@ -112,17 +112,28 @@ func (a *tokenAuthenticator) Authenticate(req *http.Request) (*AuthenticatorResp
 		return nil, errors.Wrapf(ErrMustAuthenticate, "clusterID does not match")
 	}
 
+	// If the auth provider is specified make sure it exists and enabled.
+	if token.AuthProvider != "" {
+		disabled, err := providers.IsDisabledProvider(token.AuthProvider)
+		if err != nil {
+			return nil, errors.Wrapf(ErrMustAuthenticate, "provider %s doesn't exist", token.AuthProvider)
```

Comment:
Is the only error possible that it doesn't exist?

Also, it would be good to drop the usage of the now-unmaintained pkg/errors package?

Reply:
Yep https://github.com/rancher/rancher/blob/508133fdb8beb67f9c4436d65e7ba661b987a43b/pkg/auth/providers/providers.go#L38-L45

Reply:
> Also, it would be good to drop the usage of the now-unmaintained pkg/errors package?

Agreed, but I would do that in it's own PR and clean it up everywhere at once.